File: src/tarefa-3/Empresa.ts
import { Funcionario } from "./Funcionario";

export class Empresa {
  public funcionarios: Funcionario[] = [];
  
  constructor(
    public readonly id: string,
    public readonly nome: string
  ) {}

  public AdicionarFuncionario(funcionario: Funcionario) {
    this.funcionarios.push(funcionario);
  }
}

File: src/tarefa-3/Empresa.cs
using System;
using System.Collections.Generic;

namespace EngenhariaSoftwareII.Models
{
    public class Empresa : BaseModel
    {
        public Empresa(Guid id, string nome)
        {
            Id = id;
            Nome = nome;
            Funcionarios = new List<Funcionario>();
        }

        public string Nome { get; private set; }
        public List<Funcionario> Funcionarios { get; private set; }

        public void AdicionarFuncionario(Funcionario funcionario)
        {
            Funcionarios.Add(funcionario);
        }
    }
}
